trpcultivate_phenotypes/tests/src/Kernel/Display/DisplayValidationResultWindowTest.php

```diff
@@ -0,0 +1,406 @@
+                      ['Row #14 - Value #1', 'Row #14 - Value #2', 'Row #14 - Value #3'],
+                      ['Row #15 - Value #1', 'Row #15 - Value #2', 'Row #15 - Value #3'],
+                    ],
+                  ],
+                ],
+              ],
+            ],
+          ],
+        ],
+        ['tcp-validate-fail'],
+      ],
+      // #7: An image.
+      [
+        'an image',
+        [
+          [
+            'title' => 'Validation Title Text - Todo',
+            'status' => 'todo',
+            'details' => [
+              '#theme' => 'image_style',
+              '#style_name' => 'thumbnail',
+              '#uri' => __DIR__ . '/../Fixtures/png.png',
+            ],
+          ],
+        ],
+        ['tcp-validate-todo'],
+      ],
+      // #8: A link type Drupal render array.
+      [
+        'a link',
+        [
+          [
+            'title' => 'Validation Title Text - Todo',
+            'status' => 'todo',
+            'details' => [
+              '#type' => 'link',
+              '#title' => 'KnowPulse',
+              '#url' => 'knowpulse.usask.ca',
+            ],
+          ],
+        ],
+        ['tcp-validate-todo'],
+      ],
+      // Pass, fail and todo concurrent validation items.
+      [
+        'concurrent validation items',
+        [
+          [
+            'title' => 'Validation Title Text - Pass',
+            'status' => 'pass',
+            'details' => [],
+          ],
+          [
+            'title' => 'Validation Title Text - Fail',
+            'status' => 'fail',
+            'details' => [
+              '#type' => 'item',
+              '#title' => 'Validator Case Message',
+              'items' => [
+                '#theme' => 'item_list',
+                '#type' => 'ul',
+                '#items' => [
+                  'Failed Item #1',
+                ],
+              ],
+            ],
+          ],
+          [
+            'title' => 'Validation Title Text - Todo',
+            'status' => 'todo',
+            'details' => [],
+          ],
+        ],
+        [
+          'tcp-validate-pass',
+          'tcp-validate-fail',
+          'tcp-validate-todo',
+        ],
+      ],
+      // #10: Malformed render array.
+      [
+        'malformed render array',
+        [
+          [
+            'title' => 'Validation Title Text - Fail',
+            'status' => 'fail',
+            'details' => [
+              '#not_my_type' => 'not_a_type',
+              '#not_a_property' => 'some property',
+              '#markup' => 'Lorem ipsum dolor sit amet',
+              'table' => [
+                '#type' => 'table',
+                '#captionzzzz' => 'Validator Case Message',
+                '#header-array' => ['Header 1', 'Header 2'],
+                '#row' => [
+                  ['Row #1 - Value #1', 'Row #1 - Value #2', 'Row #1 - Value #3', 'Row #1 - Value #4'],
+                ],
+              ],
+            ],
```

Comment:
This is not really a malformed render array. I would instead do something like this 😈 
```suggestion
            'details' => new \stdClass(),
```

Reply:
With the 2nd parameter type of the test method set to array this case triggers a type error.
![image](https://github.com/user-attachments/assets/48164957-6041-44cb-a589-328514878faf)

Reply:
That error is being thrown by your test though, not Drupal or our module. As such it indicates an assumption in the testing environment 🤔

Reply:
The 😈 in the details... I tried this case and nothing is rendered 
<img width="261" alt="image" src="https://github.com/user-attachments/assets/e46d9336-e429-4ef9-8060-0ae39a9de8a7">

Reply:
When I tried Lacey's suggestion through the UI I got this:
<img width="1386" alt="Screenshot 2024-11-22 at 4 01 54 PM" src="https://github.com/user-attachments/assets/982b13b1-37ad-43b9-92f9-6bff8b36aa8d">

So I think the easier way to test this scenario is to test it in a separate test method, and there be a try-catch to catch the Exception.

Reply:
Let's just remove this scenario entirely. Instead we can add a separate test to check exceptions.